pkg/logs/input/tailer/tailer_windows.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build windows
+
+package tailer
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/decoder"
+)
+
+func (t *Tailer) startReading(offset int64, whence int) error {
+	var err error
+	t.fullpath, err = filepath.Abs(t.path)
+	if err != nil {
+		t.source.Tracker.TrackError(err)
+		return err
+	}
+	log.Info("Opening ", t.fullpath)
+	t.readOffset = offset
+	t.decodedOffset = offset
+
+	go t.readForever()
+	return nil
+}
+
+func (t *Tailer) readAvailable() (err error) {
+	err = nil
+	f, err := os.Open(t.fullpath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	st, err := f.Stat()
+	if err == nil {
+		sz := st.Size()
+		log.Debugf("Size is %d, offset is %d", sz, t.GetReadOffset())
+		if sz == 0 {
+			log.Debug("File size now zero, resetting offset")
+			t.SetReadOffset(0)
+			t.SetDecodedOffset(0)
+		} else if sz < t.GetReadOffset() {
+			log.Debug("Offset off end of file, resetting")
+			t.SetReadOffset(0)
+			t.SetDecodedOffset(0)
+		}
+	} else {
+		log.Debugf("Error stat()ing file %v", err)
+	}
+	f.Seek(t.GetReadOffset(), io.SeekStart)
+	inBuf := make([]byte, 4096)
+
+	for {
+		n, err := f.Read(inBuf)
+		if n == 0 || err != nil {
```

Comment:
I think you should return the error to catch `EOF`

Reply:
good point